### ISSUE TITLE
Docu

### ISSUE THREAD

**Issue by clairelee0817** [michellekuruvilla/healthmedia#56]
---

### Purpose
- Allow users to upload an image.
- Display the image preview.
- Send image to backend for prediction.
- Display prediction results and rating.
- Keep a history of predictions with option to export CSV.

### Key Features
- Upload and preview photos.
- Button to trigger prediction request.
- Table showing historical predictions stored in localStorage.
- CSV export and clear history buttons.
- Responsive and styled with animations (AOS library).

### Workflow
1. User selects an image file.
2. Image preview is shown.
3. User clicks **Predict**.
4. Image file sent as `POST` request to `/api/predict-likes`.
5. Prediction results received as JSON.
6. Results displayed, saved in localStorage, and shown in table.

---

### Technology
- Python Flask server
- Machine learning model (e.g., scikit-learn) pre-trained on Legoland Instagram data

### Data and Model
- Dataset: `datasets/legoland.csv` containing image feature columns (Brightness, Saturation, Size) and `numLikes`.
- Model trained to predict `numLikes` from image features.

#### Request
- Method: POST
- Content-Type: `multipart/form-data`
- Field: `image` (the uploaded photo)

#### Response
- JSON object containing:
  - `brightness`: extracted brightness value
  - `saturation`: extracted saturation value
  - `size`: image size in megapixels
  - `predicted_likes`: predicted likes count
  - `rating_score`: percentage relative to dataset average likes
  - `rating_label`: qualitative rating ("Excellent", "Good", "Moderate", "Poor")

#### Process
1. Receive uploaded image.
2. Save image temporarily.
3. Extract features (`brightness`, `saturation`, `size`).
4. Use ML model to predict likes.
5. Calculate rating score relative to average dataset likes.
6. Return JSON with all relevant data.
7. Delete temporary image file.

---

### `extract_image_features(filepath)`
- Loads image from `filepath`.
- Calculates:
  - **Brightness:** average intensity of pixels.
  - **Saturation:** average saturation from HSV or HSL color space.
  - **Size:** total pixels converted to megapixels.

### `average_likes()`
- Computes average likes from dataset if available.
- Returns a mock value (e.g., 100) if dataset missing.

### `classify_rating(score)`
- Classifies rating score into labels:
  - ≥125%: Excellent
  - ≥115%: Good
  - ≥100%: Moderate
  - <100%: Poor

---

### Prerequisites
- Python 3.x environment
- Flask
- pandas, numpy, scikit-learn (or equivalent)
- Trained ML model saved as `model.pkl`
- Dataset file `datasets/legoland.csv`

### Backend Setup
1. Install dependencies:
   ```bash
   pip install flask pandas numpy scikit-learn
